jq.c

```diff
@@ -308,62 +409,52 @@ jdbc_jvm_init(const ForeignServer * server, const UserMapping * user)
 
 	if (FunctionCallCheck == false)
 	{
-		classpath = (char *) palloc0(strlen(strpkglibdir) + 19);
-		snprintf(classpath, strlen(strpkglibdir) + 19, "-Djava.class.path=%s", strpkglibdir);
-
 		if (opts.maxheapsize != 0)
 		{						/* If the user has given a value for setting
 								 * the max heap size of the JVM */
-			options = (JavaVMOption *) palloc0(sizeof(JavaVMOption) * 2);
+			options = (JavaVMOption *) palloc0(sizeof(JavaVMOption));
 			maxheapsizeoption = (char *) palloc0(sizeof(int) + 6);
 			snprintf(maxheapsizeoption, sizeof(int) + 6, "-Xmx%dm", opts.maxheapsize);
-			options[0].optionString = classpath;
-			options[1].optionString = maxheapsizeoption;
-			vm_args.nOptions = 2;
+			options[0].optionString = maxheapsizeoption;
+			vm_args.nOptions = 1;
 		}
 		else
 		{
-			options = (JavaVMOption *) palloc0(sizeof(JavaVMOption));
-			options[0].optionString = classpath;
-			vm_args.nOptions = 1;
+			vm_args.nOptions = 0;
 		}
-		vm_args.version = JNI_VERSION_1_2;
+		vm_args.version = JNI_VERSION;
 		vm_args.options = options;
 		vm_args.ignoreUnrecognized = JNI_FALSE;
 
 		/* Create the Java VM */
 		res = JNI_CreateJavaVM(&jvm, (void **) &Jenv, &vm_args);
-		if (res < 0)
+        if (res == JNI_EEXIST) {
+            ereport(DEBUG3, errmsg("Java VM has already been created. "
+                        "Re-use the existing Java VM."));
+            res = JNI_GetCreatedJavaVMs(&jvm, 1, NULL);
```

Comment:
Just for clarification, this reuses an existing JVM without using specified options including max heap size. So, it could cause an unexpected JVM heap size issue or something if the original JVM has been created with an unsuitable options?

Reply:
That is a good point. As far as I know, we cannot modify the heap size of the running JavaVM. Therefore, we must ignore the heap size setting when re-using the existing JavaVM, and users must be responsible for setting the appropriate heap sizes for both `jdbc_fdw` and another extension that uses JNI. One thing we can do is put a warning in the log message. Do you have any idea about this?

Reply:
> One thing we can do is put a warning in the log message

When an actual JVM heap size is different from a specified parameter, right? It sounds a good idea. Users can notice the difference thanks to the message.

Reply:
> When an actual JVM heap size is different from a specified parameter, right? 
Correct. By the way, do you know how to get the allocated heap size with `-Xmx` from the running JavaVM? `Runtime.maxMemory()` seems to work for this purpose, but I'm not sure if this matches exactly with `-Xmx`.

Reply:
I see. I don't come up with a good way... (`JNI_GetDefaultJavaVMInitArgs` doesn't work for the purpose, right)

Reply:
Right. `JNI_GetDefaultJavaVMInitArgs` return a default value of the  `JavaVMInitArgs` type. If it is impossible to get the exact size of `-Xmx` from the running JNI JavaVM, then what we can do would be to always show a warning message regardless of the current heap size.

Reply:
@komamitsu I've pushed one commit that adds a log message showing the current max heapsize (got with `Runtime.maxMemory()`) of the existing JavaVM. https://github.com/scalar-labs/jdbc_fdw/pull/1/commits/1f7ef768caa1b02d12ed2f1c9da414ad49ab9bae Could you let me know what you think of it? If it is too complicated, I would revert it.

Reply:
Sounds good to me! I left a minor comment.